### DISCUSSION TITLE
Post Your Build Photos Here

### DISCUSSION THREAD

**Discussion by nebhead** [nebhead/PiTender#5] [Show and tell]
I thought it would be fun to have a section of build photos to show off your build!

**Comment by nebhead** [comment]
Getting this started, my friend built up this to prototype the system.  

![messages_0](https://user-images.githubusercontent.com/19876886/111084931-26917e80-84d2-11eb-9739-699ead5e565b.jpeg)
![messages_0 (1)](https://user-images.githubusercontent.com/19876886/111084932-28f3d880-84d2-11eb-8ff1-6d9b024c4890.jpeg)
![messages_0 (2)](https://user-images.githubusercontent.com/19876886/111084933-298c6f00-84d2-11eb-8b69-21fbeda24559.jpeg)
![messages_0 (3)](https://user-images.githubusercontent.com/19876886/111084934-298c6f00-84d2-11eb-8bd4-26f22227190b.jpeg)

**Comment by Jannikovic145** [comment]
Okay, so then I would like to show you what I've done, based on your great coding idea!

![876FB0A1-4D26-46AA-BF52-DEABBA718782](https://user-images.githubusercontent.com/43824759/111200229-cc6fe680-85c1-11eb-89e0-77f1cf0f3c68.JPG)
![BA643ABB-9EF6-486D-B299-A570D96ACBC0](https://user-images.githubusercontent.com/43824759/111200266-d560b800-85c1-11eb-8e1d-dccedbf72b78.JPG)
![IMG_9653](https://user-images.githubusercontent.com/43824759/111200302-e1e51080-85c1-11eb-99c4-6330176c5f98.jpg)
![IMG_9654](https://user-images.githubusercontent.com/43824759/111200352-ec9fa580-85c1-11eb-8d73-41d22ebe319d.jpg)
![IMG_9655](https://user-images.githubusercontent.com/43824759/111200380-f923fe00-85c1-11eb-858b-6aa8ce94d991.jpg)
![IMG_9656](https://user-images.githubusercontent.com/43824759/111200402-00e3a280-85c2-11eb-9c57-b162bf5e762b.jpg)

**Comment by maddog0487** [comment]
![IMG_20230221_153004_01](https://user-images.githubusercontent.com/126736174/235513734-0773e0d3-7e9a-47f4-8e92-f3d288a7c092.jpg)

Here is the pump assembly, I used 12 pumps, this will be mounted inside my enclosure.  More pictures to follow, I added a homemade thermoelectric cooler to keep the mixes cold and an alcohol chilling system using a laboratory condenser... (hope it works!)

**Comment by maddog0487** [comment]
![20230501_185242](https://user-images.githubusercontent.com/126736174/235547113-213ba188-3fb5-4324-ab40-3bc666d020f3.jpg)
![20230501_185316](https://user-images.githubusercontent.com/126736174/235547120-02893e3c-dfb6-4743-a5cf-48927824ebd8.jpg)
![20230501_185354](https://user-images.githubusercontent.com/126736174/235547129-a63c3129-68ce-4618-96bf-a2085e586887.jpg)
![20230501_185044](https://user-images.githubusercontent.com/126736174/235547136-1b5305d7-4971-4610-8408-904734e981cc.jpg)
![20230501_185100](https://user-images.githubusercontent.com/126736174/235547154-2d15f9fa-47ee-4199-bddf-df4ad8febd54.jpg)
![20230501_185132](https://user-images.githubusercontent.com/126736174/235547161-f27e75cf-f1e6-47bc-abfe-ff01a70b48fa.jpg)
 Ok, here are some more pictures, the panel shows the status of my different power supplies with a separate USB power supply, the display is to monitor the current draw to make sure the chiller are all working.  I just placed the pump assembly approximately where it is going to be, the center board is a manual control I whipped up for priming / cleaning.  I plan on having 6 spirits and 5 chilled mixes, not sure how the cooler is going to respond so added a variable fan speed control and temperature display / controller to switch between series and parallel operation.   Above the pump assembly is a propylene glycol pump with a chiller on the back to pump threw the condenser, spirits go in the top; down below is a funnel with 5 inlets for the mixes.  Hoping for a live test in the coming weeks, still have a few pcb assemblies to configure and testing before its' ready.

**Reply by nebhead** [reply]
I mean... this takes the cake.  What an impressive machine.  Nice job - best of luck getting it all going!

**Comment by maddog0487** [comment]
![20230522_174243](https://github.com/nebhead/PiTender/assets/126736174/40d7e8df-be95-4dce-aa84-365847db2ba9)

Well, took it for a test drive last week and made a Bourbon Sour, unfortunately I can't upload a video but it looks pretty cool a night.  I wasn't happy with the pre-chiller so I decided to make my own bar fridge using the cooling unit I made and drilled a hole in the top to shorten the length of tubes required plus keep the ingredients cooler.  Today I was working in the yard making Sparkling Cranberry Lemonade.  I am sure there are more mods to come but so far so good.  (Fixing the "adding a recipe issue" would be good...)  Thanks again to all of those that worked on this project!

**Reply by nebhead** [reply]
Amazing!  I'll look into fixing the adding a recipe issue this week.  See if I can remember how it all worked.  This project was one of the first where I was learning JS, so there are bound to be bugs.  :P

**Comment by maddog0487** [comment]
![20230514_211948](https://github.com/nebhead/PiTender/assets/126736174/5c779250-d9f9-491f-b899-fae477995037)
![20230514_211914](https://github.com/nebhead/PiTender/assets/126736174/c2a3f83c-5adc-41d9-ace9-961575f77f03)
Couple of more photos showing a more completed phase of construction with the cooling system, I want to work on this next to shorten the hoses and get better efficiency, can only get the temperature down to around 3C, uses propylene glycol coolant which apparently is a food additive.

**Reply by nebhead** [reply]
This build is truly epic!  Well done indeed!  This is going to be a hit at parties.  Love it!

**Comment by maddog0487** [comment]
![20230628_075806](https://github.com/nebhead/PiTender/assets/126736174/dfff9944-08ae-4d35-980d-c09cf3b85c74)
More insanity!  I upgraded the alcohol chiller system to 5 Peltier modules for about 380W of cooling, of course more heat so I added a dual radiator cooling system which seems to like fans, 6 x 120mm fans plus a bunch (12) more I had laying around, it is in my garage so ambient plays a huge factor in maximum cooling differential; with ambient around 21C I can reach -10C, even at higher ambient and -6C on the chiller I can cool 2 oz. of room temperature liquor (23C) to 4C in a single pass.

The relays used for the chiller were supposed to handle 40A....I was only running at 25A but they got hot and I was losing about 2V so I added the flux capacitor, parallel connected N-channel MOSFETs to bring the volt drop down to less than 100mV, also the econo relays I purchased from the "jungle" had maybe a 16AWG wiring harness, I beefed everything up to 10AWG, boy that cab-ty got expensive! AND when I improved the switching the Peltier modules drew way more power so I had to add a 3rd power supply for the fridge section.

So next story, the fridge wasn't meeting my target 4C so I added a water block and MOSFET switch as well with future linear voltage control (reviving my tech report from college circa 1992), and beefed up the wire...running a test now but dropped 1L of water in a glass jar from 32C to 5C in about 5 hours, the ambient is rising so taking a little longer than expected but still a pretty decent cooling rate, I may need to add an additional cooling block or another module or both plus some thermal mass to buffer the temperature rise during the day.

Oh, one last thing, added in a T line and another pump to the code providing the option of recirculating through the chiller or another ingredient plus the fancy window for better wifi performance.

**Reply by nebhead** [reply]
WOW!!  You are definitely living up to your username. :) I'd say that this build is BY FAR the most ambitious and involved ever!   Very nice work and very cool stuff (pun intended).

**Comment by maddog0487** [comment]
![20230710_205008](https://github.com/nebhead/PiTender/assets/126736174/1c56caa7-1a38-4615-a518-b0d766e5cb3e)
![20230802_180148 (1)](https://github.com/nebhead/PiTender/assets/126736174/17a55e8a-2117-4975-a3b8-a60a4c1ed832)

Haven't posted in a while but here are the last modifications, I added the analog temperature controller and a feedback to show the voltage on the peltier coolers, works preety good but maybe thinking about using a Pi for monitoring and control in the future, still have a ways to go in my programming to add it to your code... another project!  Haven't decided on the fate of this project, possibly rebuild into a Pelican case so it is transportable but also thinking about "Martini Mode", goal is to dispense a drink at -10C kicking a few ideas around but nothing is sticking yet.  I did lose my cooling pump a couple of weeks back, hopefully the more expensive "real" pump will last longer than a few months of intermittant use, good thing I used computer power supplies with monitoring that shutdown at the first sign of trouble!

**Comment by maddog0487** [comment]
![20240425_194755](https://github.com/nebhead/PiTender/assets/126736174/ea122679-d4e6-45fa-ae2f-2a5756069f15)
![20240416_202250](https://github.com/nebhead/PiTender/assets/126736174/07b16467-891e-4c79-92b8-63aba7db07bf)

Just posting my upgraded build before moving on, another enthusiastic group created a new interface that almost does too much.  Still trying to figure it out but it is easy to view and manage GPIO, Pumps, recipes, etc..  Thanks for creating the code, it has served me well.

**Reply by nebhead** [reply]
Amazing!  What is the other project - I might start sending people over there as I'm not spending time on this one anymore myself. :)

**Reply by maddog0487** [reply]
I didn't want to offend you, :)
cocktailpi.org, check it out, I joined the Discord group.  Working on documentation for my build to post.  However my code it a little dis-organized for the chiller control, works but not pretty and needs some attention.

**Reply by nebhead** [reply]
Thanks!  Not offended at all, I'm all for it!  Especially since I'm not maintaining this project anymore.  

CocktailPi looks like a great project.  I wonder if any of it was inspired by PiTender. :)  Keep making and have fun!

**Reply by maddog0487** [reply]
I am 100% positive it was inspired by your build!  Again, thanks.
